import-service/src/__tests__/importFileParser.test.ts

```diff
@@ -0,0 +1,130 @@
+import { S3Event } from 'aws-lambda';
+import { S3Client, GetObjectCommand, CopyObjectCommand, DeleteObjectCommand } from '@aws-sdk/client-s3';
+import { mockClient } from 'aws-sdk-client-mock';
+import { Readable } from 'stream';
+import { handler } from '../importFileParser';
+
+const s3ClientMock = mockClient(S3Client);
+
+describe('importFileParser lambda', () => {
+  beforeEach(() => {
+    s3ClientMock.reset();
+  });
+
+  it('should process CSV file from S3', async () => {
+    // Mock S3 event
+    const mockEvent: S3Event = {
+      Records: [{
+        s3: {
+          bucket: {
+            name: 'test-bucket'
+          },
+          object: {
+            key: 'uploaded/test.csv'
+          }
+        }
+      }]
+    } as any;
+
+    // Create a mock readable stream with CSV content
+    const mockCsvContent = 'id,title,price\n1,Test Product,10.99';
+    const mockStream = Readable.from([mockCsvContent]);
+
+    // Mock S3 GetObjectCommand response
+    s3ClientMock
+      .on(GetObjectCommand).resolves({
+        Body: mockStream
+      })
+      .on(CopyObjectCommand).resolves({})
+      .on(DeleteObjectCommand).resolves({});
+
+    // Mock console.log to verify output
+    const consoleSpy = jest.spyOn(console, 'log');
+
+    await handler(mockEvent);
+
+    // Verify S3 client was called with correct parameters
+    // Verify all S3 operations were called
+    expect(s3ClientMock.calls()).toHaveLength(3); // Get, Copy, and Delete operations
+
+    // Verify GetObjectCommand
+    const getObjectCall = s3ClientMock.commandCalls(GetObjectCommand)[0];
+    expect(getObjectCall.args[0].input).toEqual({
+      Bucket: 'XXXXXXXXXXX',
+      Key: 'uploaded/test.csv'
+    });
+
+    // Verify CopyObjectCommand
+    const copyObjectCall = s3ClientMock.commandCalls(CopyObjectCommand)[0];
+    expect(copyObjectCall.args[0].input).toEqual({
+      Bucket: 'XXXXXXXXXXX',
```

Comment:
https://github.com/slavafv/nodejs-aws-shop-be/pull/4/files#diff-c7f1e172bef2323c842840714a8e8cf233e0150cb2f1fd535a3e022bdc31509cR20